che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/service/add/wizard/CreateServiceWizard.java

```diff
@@ -0,0 +1,219 @@
+                                     for (DeploymentConfig config : configs) {
+                                         List<Container> containers = config.getSpec().getTemplate().getSpec().getContainers();
+
+                                         if (containers == null) {
+                                             config.getSpec().getTemplate().getSpec().setContainers(containersForUpdate);
+                                             continue;
+                                         }
+
+                                         for (Container containerForUpdate : containersForUpdate) {
+                                             for (Container container : containers) {
+                                                 updateContainerEnv(container, containerForUpdate.getEnv());
+                                             }
+                                         }
+                                         config.getMetadata().getLabels().put("database", deploymentConfigName);
+                                     }
+                                     return configs;
+                                 }
+                             }).thenPromise(updateApplicationDeploymentConfigs());
+            }
+        };
+    }
+
+    private Function<List<DeploymentConfig>, Promise<JsArrayMixed>> updateApplicationDeploymentConfigs() {
+        return new Function<List<DeploymentConfig>, Promise<JsArrayMixed>>() {
+            @Override
+            public Promise<JsArrayMixed> apply(List<DeploymentConfig> configs) throws FunctionException {
+                List<Promise<?>> promiseList = new ArrayList<>();
+                for (DeploymentConfig config : configs) {
+                    Promise<DeploymentConfig> promise = client.updateDeploymentConfig(config);
+                    promiseList.add(promise);
+                }
+                Promise<?>[] promises = promiseList.toArray(new Promise<?>[promiseList.size()]);
+                return Promises.all(promises);
+            }
+        };
+    }
+
+    private void updateContainerEnv(Container container, List<EnvVar> envVariables) {
+        List<EnvVar> containerEnvs = container.getEnv();
+
+        if (containerEnvs == null) {
+            container.setEnv(envVariables);
+            return;
+        }
+
+        Map<String, EnvVar> envNamesToEnv = new HashMap<>();
+        for (EnvVar containerEnv : containerEnvs) {
+            envNamesToEnv.put(containerEnv.getName(), containerEnv);
+        }
+
+        for (final EnvVar envVarForEdit : envVariables) {
+            String name = envVarForEdit.getName();
+            if (envNamesToEnv.containsKey(name)) {
+                containerEnvs.remove(envNamesToEnv.get(name));
+            }
+            containerEnvs.add(envVarForEdit);
+        }
+    }
+
+    private Operation<PromiseError> handleError(final CompleteCallback callback) {
```

Comment:
I prefer to write handing of error in place without private method if it calls in one place